system_metrics_collector/CMakeLists.txt

```diff
@@ -30,29 +30,65 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(metrics_statistics_msgs REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 find_package(rcpputils REQUIRED)
 find_package(rcutils REQUIRED)
 
-add_library(system_metrics_collector SHARED
+add_library(system_metrics_collector_base STATIC
```

Comment:
Currently, if `libsystem_metrics_collector_process_cpu_component.so` depends on a `libsystem_metrics_collector_base.so`, it can't find it at runtime.

I can try setting `rpath`. Setting `rpath` to a relative path is discouraged. I'll set it to an absolute path, but that means these `.so` files can't be moved elsewhere after being installed.

Reply:
From the design review: rename this to lib_system_metrics_collector

Reply:
Done.

We're going with just one monolithic shared library that provides multiple components.